tuna/rocmlir/export_configs.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+###############################################################################
+#
+# MIT License
+#
+# Copyright (c) 2022 Advanced Micro Devices, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+###############################################################################
+""" Module for exporting configs for use in performance runs """
+
+from tuna.parse_args import TunaArgs, setup_arg_parser
+from tuna.rocmlir.rocmlir_tables import RocMLIRDBTablesConv, RocMLIRDBTablesGEMM
+
+
+def main():
+  """Import conv-configs file into database rocmlir_conv_config table."""
+  # pylint: disable=duplicate-code
+  parser = setup_arg_parser('Export perf-configs from MySQL db',
+                            [TunaArgs.VERSION, TunaArgs.SESSION_ID])
+  parser.add_argument('-f',
+                      '--file_name',
+                      type=str,
+                      dest='file_name',
+                      help='File to import')
+  parser.add_argument(
+      '--config_type',
+      dest='config_type',
+      help='Specify configuration type',
+      default='convolution',
+      choices=['convolution', 'gemm'],  # +++pf: eventually an Enum
+      type=str)
+  parser.add_argument('--append',
+                      dest='append',
+                      action='store_true',
+                      help='Append to file instead of overwriting')
+  args = parser.parse_args()
+  if args.config_type == "convolution":
```

Comment:
More on this later.
1.  RocMLIRDBTables is subclassed by operation.
2. The export method lives on the subclass.
3. For now, we're writing the same kind of .tsv file as the existing non-Tuna tuning process.

Reply:
Yes, miopen only has MIOpenDBTables. Depending on the config_type argument to the constructor here, miopen would switch to either convolution or batchnorm. The tablenames used in class are the same, but they would direct to different tables.
If you can't keep the dbt objects in the same class, Saud's suggestion of a function for these 4 lines might be a good compromise.

Reply:
Looks like miopen doesn't have functions that want to vary by operation;  rather, it has a small number of places that check config_type at runtime.  I didn't want to do that, the differences are too large and I didn't want IFs with large bodies.  I'm moving the op-specific functions into the config classes, which will get me back to one DBTables class.  On one hand, it's a little odd because they aren't table functions, but on the other I've already done it with config_string.